### PR TITLE
fix(components): [rate] fast move mouseLeave not triggered

### LINKED ISSUE #9567: [Component] [rate] el-rate 评分组件bug
Bug Type: **`Component`**

## Environment

- Vue Version: `3.2.32`
- Element Plus Version: `2.1.8`
- Browser / OS: `Chrome`
- Build Tool: `Vite`

### Related Component

- `el-rate`

### Reproduction Link

[Element Plus Playground](https://element-plus.run/#eyJBcHAudnVlIjoiPHNjcmlwdCBzZXR1cCBsYW5nPVwidHNcIj5cbmltcG9ydCB7IHJlZiwgdmVyc2lvbiBhcyB2dWVWZXJzaW9uIH0gZnJvbSAndnVlJ1xuaW1wb3J0IHsgdmVyc2lvbiBhcyBFcFZlcnNpb24gfSBmcm9tICdlbGVtZW50LXBsdXMnXG5pbXBvcnQgeyBFbGVtZW50UGx1cyB9IGZyb20gJ0BlbGVtZW50LXBsdXMvaWNvbnMtdnVlJ1xuXG5jb25zdCBtc2cgPSByZWYoJ0hlbGxvIFdvcmxkIScpXG5cbmNvbnN0IHZhbHVlMSA9IHJlZihudWxsKVxuY29uc3QgdmFsdWUyID0gcmVmKG51bGwpXG5jb25zdCB2YWx1ZTMgPSByZWYobnVsbClcbjwvc2NyaXB0PlxuXG48dGVtcGxhdGU+XG4gIDxoMT57eyBtc2cgfX08L2gxPlxuICA8ZWwtaW5wdXQgdi1tb2RlbD1cIm1zZ1wiIC8+XG5cbiAgPHA+XG4gICAgPGVsLWljb24gY29sb3I9XCJ2YXIoLS1lbC1jb2xvci1wcmltYXJ5KVwiPjxFbGVtZW50UGx1cyAvPjwvZWwtaWNvbj5cbiAgICBFbGVtZW50IFBsdXMge3sgRXBWZXJzaW9uIH19ICsgVnVlIHt7IHZ1ZVZlcnNpb24gfX1cbiAgICBcbiAgICBcbiAgICA8ZGl2IGNsYXNzPVwiZGVtby1yYXRlLWJsb2NrXCI+XG4gICAgICAgIDxzcGFuIGNsYXNzPVwiZGVtb25zdHJhdGlvblwiPkRlZmF1bHQ8L3NwYW4+XG4gICAgICAgIDxlbC1yYXRlIHYtbW9kZWw9XCJ2YWx1ZTFcIiAvPlxuICAgICAgPC9kaXY+XG4gICAgICA8ZGl2IGNsYXNzPVwiZGVtby1yYXRlLWJsb2NrXCI+XG4gICAgICAgIDxzcGFuIGNsYXNzPVwiZGVtb25zdHJhdGlvblwiPkNvbG9yPC9zcGFuPlxuICAgICAgICA8ZWwtcmF0ZSB2LW1vZGVsPVwidmFsdWUyXCIgLz5cbiAgICAgIDwvZGl2PlxuICAgICAgPGRpdiBjbGFzcz1cImRlbW8tcmF0ZS1ibG9ja1wiPlxuICAgICAgICA8c3BhbiBjbGFzcz1cImRlbW9uc3RyYXRpb25cIj5Db2xvcjI8L3NwYW4+XG4gICAgICAgIDxlbC1yYXRlIHYtbW9kZWw9XCJ2YWx1ZTNcIiAvPlxuICAgICAgPC9kaXY+XG5cbiAgICAgIHt7IHZhbHVlMSB9fSAtIHt7IHZhbHVlMiB9fSAtIHt7IHZhbHVlMyB9fVxuICA8L3A+XG48L3RlbXBsYXRlPlxuIiwiaW1wb3J0X21hcC5qc29uIjoie1xuICBcImltcG9ydHNcIjoge31cbn0iLCJfbyI6e319)

### Steps to reproduce

界面上常见三个el-rate组件；分别绑定三个不同的值，鼠标快速划过；el-rate值被选中，但是绑定的值未被赋予新值

### What is Expected?

el-rate界面表现选中和绑定的值一致

### What is actually happening?

不一致

### Additional comments

_(empty)_

image

### PR DESCRIPTION
**Pull request by chenxch**
close #9567

- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

Thanks @holazz  for the inspiration.

### CODE REVIEW

**Comment by chenxch** [comment]
cc @sxzz  I need your help to see the type error of markraw.

**Comment by sxzz** [comment]
Wait for PR https://github.com/vuejs/core/pull/6380

File: packages/components/rate/src/rate.vue [hunk 1]
```diff
@@ -147,34 +148,36 @@ const decimalStyle = computed(() => {
     width,
   }
 })
-const componentMap = computed(() =>
-  isArray(props.icons)
+const componentMap = computed(() => {
+  const icons = markRaw(
+    isArray(props.icons) ? [...props.icons] : { ...props.icons }
+  )
```
Comment (sxzz) [inline packages/components/rate/src/rate.vue:154]:
```suggestion
  let icons = isArray(props.icons) ? [...props.icons] : { ...props.icons }
  icons = markRaw(icons)
```

Comment (chenxch) [reply packages/components/rate/src/rate.vue]: Thanks bro.

**Review by holazz — APPROVED** [review]

File: packages/components/rate/src/rate.vue [hunk 2]
```diff
@@ -30,10 +30,10 @@
           ns.is('active', item <= currentValue),
         ]"
       >
-        <component
-          :is="iconComponents[item - 1]"
-          v-if="!showDecimalIcon(item)"
-        />
+        <template v-if="!showDecimalIcon(item)">
+          <component :is="activeComponent" v-show="item <= currentValue" />
```
Comment (xiaoxian521) [inline packages/components/rate/src/rate.vue:34]: Is it more appropriate to use this code replacement? 🤔
```
<component :is="item <= currentValue ? activeComponent : voidComponent" />
```

Comment (chenxch) [reply packages/components/rate/src/rate.vue:34]: The original cause of the problem was caused by component switching, destruction and re-rendering.

**Pull request merged**